src/server/game/Storages/WhoListStorage.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+* Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "WhoListStorage.h"
+#include "World.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "GuildMgr.h"
+#include "WorldSession.h"
+
+WhoListStorageMgr* WhoListStorageMgr::instance()
+{
+    static WhoListStorageMgr instance;
+    return &instance;
+}
+
+void WhoListStorageMgr::Update()
+{
+    // clear current list
+    _whoListStorage.clear();
+    _whoListStorage.reserve(sWorld->GetPlayerCount()+1);
+
+    boost::shared_lock<boost::shared_mutex> lock(*HashMapHolder<Player>::GetLock());
+
+    HashMapHolder<Player>::MapType const& m = ObjectAccessor::GetPlayers();
+    for (HashMapHolder<Player>::MapType::const_iterator itr = m.begin(); itr != m.end(); ++itr)
+    {
+        if (!itr->second->FindMap() || itr->second->GetSession()->PlayerLoading())
+            continue;
+
+        std::string pname = itr->second->GetName();
+        std::wstring wpname;
+        if (!Utf8toWStr(pname, wpname))
+            continue;
+        wstrToLower(wpname);
+
+        std::string gname = sGuildMgr->GetGuildNameById(itr->second->GetGuildId());
+        std::wstring wgname;
+        if (!Utf8toWStr(gname, wgname))
+            continue;
+        wstrToLower(wgname);
+
+        std::string aname;
+        if (AreaTableEntry const* areaEntry = sAreaTableStore.LookupEntry(itr->second->GetZoneId()))
+            aname = areaEntry->area_name[sWorld->GetDefaultDbcLocale()];
```

Comment:
Do you really need to cache the name at all? Just cache the ID.
Its already just 2 array lookups which are constant and lockless. (not even a hashmap lookup). Caching probably gives little to no benefit. Or then I just dont see something here.